content/en/blog/_posts/2022-04-18-volume-expansion-ga.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: blog
+title: "Volume expansion going GA in v1.24"
+date: 2022-04-18
+slug: volume-expansion-ga
+---
+
+**Author:** Hemant Kumar
+
+## Introduction
+
+Volume expansion was introduced as a alpha feature in Kubernetes 1.8 and it went beta in 1.11 and with Kubernetes 1.24 we are excited to announce general availability(GA)
+of volume expansion feature.
+
+This feature allows Kubernetes users to simply edit their `PersistentVolumeClaim` objects and specify new size in PVC Spec and Kubernetes will automatically expand the volume
+using storage backend and also expand the underlying file system in-use by the Pod without requiring any downtime at all if possible.
+
+
+### How to use volume expansion feature
+
+Using volume expansion is as simple as editing `storage` field of pvc spec and specifying new size. For example - given following PVC:
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: myclaim
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi --> specify new size here
+```
+
+Users can request expansion of the underlying volume by specifying a new value instead of old `1Gi` size. Once expansion is initiated - pvc's conditions can be monitored
+for completion of volume expansion operation.
+
+When Kubernetes starts expanding the volume - it will add `Resizing` condition to the PVC, which will be removed once expansion completes. More information about progress of
+expansion operation can also be obtained by monitoring events associated with pvc:
+
+```
+~> kubectl describe pvc <pvc>
+```
+
+### Storage driver support
+
+Not every volume type however is expandable by default. Some volume types such as - intree hostpath volumes are not expandable at all. For CSI volumes - the CSI driver
+must have capability `EXPAND_VOLUME` in controller or node service (or both if appropriate). For CSI volumes - please refer to documentation of your CSI driver, to find out
+if it supports volume expansion.
+
+Please refer to volume expansion documentation for intree volume types which support volume expansion - [Expanding Persistent Volumes](/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims)
+
+
+In general to provide some degree of control over volumes that can be expanded, only dynamically provisioned PVCs whose storage class has `allowVolumeExpansion` parameter set to `true` are expandable.
+
+A Kubernetes admin must edit storage class object and set `allowVolumeExpansion` field to true. For example:
+
+```
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: gp2-default
```

Comment:
(nit)
```suggestion
  name: gp3-default
```
**or** use a fictional provisioner, eg `cloud.example/cloud-block-storage`

Reply:
In light of https://github.com/kubernetes/website/issues/33036, let's _not_ suggest `gp3` as the example.